src/main/java/io/confluent/connect/jdbc/source/JdbcSourceTask.java

```diff
@@ -171,56 +162,42 @@ public void stop() throws ConnectException {
     if (stop != null) {
       stop.set(true);
     }
-    if (db != null) {
-      log.debug("Trying to close database connection");
-      try {
-        db.close();
-      } catch (SQLException e) {
-        log.error("Failed to close database connection: ", e);
-      }
+    if (cachedConnectionProvider != null) {
+      cachedConnectionProvider.closeQuietly();
     }
   }
 
   @Override
   public List<SourceRecord> poll() throws InterruptedException {
-    long now = time.milliseconds();
     log.trace("{} Polling for new data");
+
     while (!stop.get()) {
-      // If not in the middle of an update, wait for next update time
-      TableQuerier querier = tableQueue.peek();
+      final TableQuerier querier = tableQueue.peek();
+
       if (!querier.querying()) {
-        long nextUpdate = querier.getLastUpdate() +
-                          config.getInt(JdbcSourceTaskConfig.POLL_INTERVAL_MS_CONFIG);
-        long untilNext = nextUpdate - now;
-        log.trace("Waiting {} ms to poll {} next", untilNext, querier.toString());
-        if (untilNext > 0) {
+        // If not in the middle of an update, wait for next update time
+        final long nextUpdate = querier.getLastUpdate() + config.getInt(JdbcSourceTaskConfig.POLL_INTERVAL_MS_CONFIG);
+        for (long untilNext = nextUpdate - time.milliseconds(); untilNext > 0; untilNext = nextUpdate - time.milliseconds()) {
+          // The purpose of the loop is to handle spurious wakeups
+          log.trace("Waiting {} ms to poll {} next", untilNext, querier.toString());
```

Comment:
Is `Thread.sleep` subject to spurious wakeups that aren't going to throw an exception? The javadocs don't seem to specify...